### PR TITLE
Refactor lighting items that use fuel

### LINKED ISSUE #71826: Carrying a candle does not move the light from where you started it
## Reproduction:

- Light candle
- Pick it up
- Walk around
- The light stays where you picked it up, does not follow
- Pocket it
- The light moves to where you pocketed it, does not follow
- drop it and drag it
- Light follows

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Fixes #71826

This does the following:
- Makes candles a subtype of flare
- Fixes candles having lighting inconsistencies
- Fixes burning items (welder, candles, flares, etc.) not causing ignition effects when held in hand
- Adds burnt flares and melted candles to maint loot and trash spawners
- Add match lighting sound when candles are lit
- Add time defines for fuel amounts and rounded them (ex. instead of 32.3333 minutes, it's now 35 minutes)
- Light sources that burn will now spawn a trash item once fuel is spent
- Light sources that burn now have a welder hitsound
- Light sources that burn can now be extinguished by a fire extinguisher (except flares)
- Light sources that burn can now be used to ignite another object on fire (ex. a lit candle can be used to light a cigarette)
- Light sources that burn and are lit now do `BURN` damage while attacking

## Why It's Good For The Game
Code is more cleaner and consistent.  Also fixes some bugs.

## Changelog
:cl:
soundadd: Candles will now use the match lighting sound when lit
soundadd: Light sources that burn will now have a welding hitsound
fix: Fix candle light behaving erratically
fix: Fix burning items that are held in hand will now have an ignition effect on the turf. (ex. lit welders in hand will now ignite plasma in the air)
balance: Light sources that burn and are lit now do `BURN` damage while attacking
balance: Light sources fuel amounts were rounded to exact numbers (ex. instead of 32.3333 minutes, it's now 35 minutes)
qol: Light items that burn can now be extinguished by a fire extinguisher (except flares), used to ignite another object on fire, and will now leave a trash item once fuel is used
qol: Add burnt flares and melted candles to trash spawners
refactor: Refactor lighting items that use fuel to be more robust
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
does the extingusher work on the flare?

**Comment by timothymtorres** [comment]
> does the extingusher work on the flare?

Yes it does.

**Comment by Serxule** [comment]
> > does the extingusher work on the flare?
> 
> Yes it does.

it shouldnt get extingushed since emergency flares are meant to be waterproof/water-resistant (also they work in space) 
"Some emergency flares are waterproof, at least for a short while. The vast majority can be considered water-resistant"

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -293,62 +294,139 @@
...
+	if(ignition(user))
 		user.visible_message(span_notice("[user] lights \the [src]."), span_notice("You light \the [src]!"))
-		force = on_damage
-		damtype = BURN
-		START_PROCESSING(SSobj, src)
 
 /obj/item/flashlight/flare/get_temperature()
 	return on * heat
 
+/obj/item/flashlight/flare/candle
```
Comment (ChungusGamer666) [inline code/game/objects/items/devices/flashlight.dm:383]: this subtype feels very evil, maybe /obj/item/flashlight/flare should be generalized to something like /obj/item/flashlight/fueled

**Comment by timothymtorres** [comment]
> it shouldnt get extingushed since emergency flares are meant to be waterproof/water-resistant (also they work in space) "Some emergency flares are waterproof, at least for a short while. The vast majority can be considered water-resistant"

Applied with https://github.com/tgstation/tgstation/pull/72146/commits/8c0c12cbac6cb030a672b60965eaf4d892541726

File: code/game/objects/items/devices/flashlight.dm (hunk 1 above)
Comment (timothymtorres) [reply code/game/objects/items/devices/flashlight.dm:392]: I already considered adding a `/obj/item/flashlight/fuel` subtype but ultimately decided it wasn't necessary.  We don't have enough items with deviating properties to make it worth it.

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -293,62 +294,151 @@
...
+/obj/item/flashlight/flare/update_brightness()
+	..()
+	inhand_icon_state = "[initial(inhand_icon_state)]" + (on ? "-on" : "")
+	update_appearance()
 
 /obj/item/flashlight/flare/process(delta_time)
 	open_flame(heat)
-	fuel = max(fuel -= delta_time, 0)
-	if(fuel <= 0 || !on)
+	fuel = max(fuel -= delta_time * (1 SECONDS), 0)
```
Comment (Kylerace) [inline code/game/objects/items/devices/flashlight.dm:352]:
```suggestion
	fuel = max(fuel - delta_time * (1 SECONDS), 0)
```

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Pull request merged**